console/src/main/scala/io/shiftleft/console/cpgcreation/CCpgGenerator.scala

```diff
@@ -1,14 +1,20 @@
 package io.shiftleft.console.cpgcreation
 
-import io.shiftleft.console.FrontendConfig
+import io.shiftleft.console.{FrontendConfig, InstallConfig}
 
-import java.nio.file.Path
+import java.nio.file.{Path, Paths}
 
 /**
   * Fuzzy C/C++ language frontend. Translates C/C++ source files
   * into code property graphs via fuzzy parsing.
   * */
-case class CCpgGenerator(config: FrontendConfig, rootPath: Path) extends CpgGenerator {
+case class CCpgGenerator(config: FrontendConfig, installConfig: InstallConfig) extends CpgGenerator {
+
+  val binaryPath: Path = {
```

Comment:
since the code for `binaryPath` is repeating, maybe it would make sense to put it inside _CpgGenerator_ and have the script name instead be specific for the various subclasses?

Reply:
Funny you should mention that. I originally did exactly that. The issue I ran into there was dealing with the front-ends with Windows options in a similar way, but 2 options there is to have a separate `winPath` for all the front-ends (which would be empty for most of them), or to hack it in for the 2 others. Instead of deciding on that now, I opted to go for the minimal change in this case, with ideas of cleaning everything up properly later.

An option I didn't consider is to define the `binaryPath` in `CpgGenerator`, but to only use that for the *nice* frontends (which is already better than what I have here.

Reply:
alrighty, if you want to deal with Windows at a later point, :+1: from my side.
the one thing I'd still adjust though is the name of the val, maybe `invocationScriptPath` is a better description?

Reply:
That's a much better name!